geopandas/geoseries.py

```diff
@@ -665,8 +665,14 @@ def select(self, *args, **kwargs):
         return self._wrapped_pandas_method("select", *args, **kwargs)
 
     @doc(pd.Series)
-    def apply(self, func, convert_dtype: bool = True, args=(), **kwargs):
-        result = super().apply(func, convert_dtype=convert_dtype, args=args, **kwargs)
+    def apply(self, func, convert_dtype: bool = None, args=(), **kwargs):
+        if convert_dtype is not None:
```

Comment:
Just for clarity since this is a bit ugly. 

In pandas 2.1, convert dtype is deprecated and the future behaviour is the same as convert_dtype = True (https://github.com/pandas-dev/pandas/pull/52786)

So this just preserves the existing signature and forwards what the user provides. If we have the default of None, we pass nothing through on pandas 2.1 to get the convert_dtype=True behaviour without the warning